docs/faq.md

```diff
@@ -38,6 +38,54 @@ targets:
           compiler: dart2js
 ```
 
+## How can I build with multiple configurations
+
+The build system supports two types of builds, "dev" and "release". By default
+with `build_runner` the "dev" version is built by default regardless of the
+command used, build in release mode by passing the `--release` flag. With
+`webdev` the default mode for the `serve` command is dev, and the default mode
+for the `build` command is release. The `build` command can use dev mode with
+the `--no-release` flag.
+
+Options can be configured per mode, and they are merged by key. The `options`
+field defines configuration used in all modes, and the `dev` and `release`
+fields defines the overrides to those defaults for the specific mode chosen.
+Builders can define their own defaults by mode which is overridden by user
+config. For example `build_web_compilers` defines options that use `dartdevc`
+compiler in dev mode, and `dart2js` in release mode.
+
+The following configuration builds with `dart2js` always, passes `--no-minify`
+in dev mode, and passed `-O3` in release mode:
+
+```yaml
+targets:
+  $default:
+    builders:
+      build_web_compilers|entrypoint:
+        options:
+          compiler: dart2js
+        dev_options:
+          dart2js_args:
+          - --no-minify
+        release_options:
+          dart2js_args:
+          - -O3
+```
+
+If you need other configurations in addition to dev and release, you can define
+multiple `build.yaml` files. For instance if you have a `build.debug.yaml` file
+you can build with `--config debug` and this file will be used instead of the
+default `build.yaml`. The dev and release flavors still apply. `pub run
+build_runner serve --config debug` will use the `dev_options` in
+`build.debug.yaml`, while `pub run build_runner build --config debug --release`
+will use the `release_options` in `build.debug.yaml`.
```

Comment:
Does this `--config` option work also with `webdev`? If so, it would be great to mention it. If not, same.

Reply:
I think  you have to put it after a `--`, so `webdev <args> -- --config debug` for instance. Webdev doesn't support all the build_runner arguments directly but you can forward additional args to build_runner that way.